pkg/build/api/v1/types.go

```diff
@@ -658,6 +734,27 @@ type BuildRequest struct {
 
 	// Env contains additional environment variables you want to pass into a builder container
 	Env []kapi.EnvVar `json:"env,omitempty"`
+
+	// BuildCauses can pass information from the controllers to the buildconfig
+	// generator about who/what has triggered a build through the request.
+	BuildCauses []*BuildTriggerCauses `json:"BuildCauses,omitempty"`
+}
+
+// BuildTriggerCauses passes information from the controllers to the
+// buildconfig generator about who/what has triggered a build
+type BuildTriggerCauses struct {
```

Comment:
We shouldn't have 3 structures for this same information.

We should have a single BuildCause structure which can be provided with the BuildRequest, and which is included in the Build object (either in the Spec or the Status.  Probably the Spec).